src/solver/libModelObject/model.cpp

```diff
@@ -19,14 +19,120 @@
 ** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
 */
 
+#include <algorithm>
+#include <iterator>
+#include <utility>
+#include <vector>
+
 #include <antares/solver/libObjectModel/model.h>
 
 namespace Antares::Solver::ObjectModel
 {
 
-std::vector<Constraint*> getConstraints()
+/**
+ * \brief Builds and returns the Model object.
+ *
+ * \return The constructed Model object.
+ */
+Model ModelBuilder::build()
+{
+    return model_;
+}
+
+/**
+ * \brief Sets the ID of the model.
+ *
+ * \param id The ID to set.
+ * \return Reference to the ModelBuilder object.
+ */
+ModelBuilder& ModelBuilder::withId(std::string_view id)
+{
+    model_.id_ = id;
+    return *this;
+}
+
+/**
+ * \brief Sets the objective of the model.
+ *
+ * \param objective The Expression object representing the objective.
+ * \return Reference to the ModelBuilder object.
+ */
+ModelBuilder& ModelBuilder::withObjective(Expression objective)
+{
+    model_.objective_ = objective;
+    return *this;
+}
+
+/**
+ * \brief Sets the parameters of the model.
+ *
+ * \param parameters A vector of Parameter objects to set.
+ * \return Reference to the ModelBuilder object.
+ *
+ * inputs it not garanteed to be valid after the call
+ */
+ModelBuilder& ModelBuilder::withParameters(std::vector<Parameter>& parameters)
+{
+    std::transform(parameters.begin(),
+                   parameters.end(),
+                   std::inserter(model_.parameters_, model_.parameters_.end()),
+                   [](/*Non const to prevent copy*/ Parameter& parameter)
+                   { return std::make_pair(parameter.Name(), std::move(parameter)); });
+    return *this;
+}
+
+/**
+ * \brief Sets the variables of the model.
+ *
+ * \param variables A vector of Variable objects to set.
+ * \return Reference to the ModelBuilder object.
+ *
+ * inputs it not garanteed to be valid after the call
```

Comment:
To better reflect this, we might as well use `std::vector<T>&&`
```cpp
void takeControl(std::vector<T>&& vec)
{
    
}
```

Reply:
Nice.
I kept the names to stay in the convention of builder but changed to rvalue ref. Client are forced to std::move(value) so no mistake possible